src/config/field-overrides/program-indicator/OrgUnitField.js

```diff
@@ -87,18 +93,40 @@ export default class OrgUnitField extends Component {
         if (this.hasRequiredParams()) {
             this.fetchDataElements();
         }
+        this.setState({ options: this.getOptions() });
 
         this.updateManualPrevProps();
     }
 
-    componentDidUpdate() {
-        if (
-            this.hasRelevantPropChanged() &&
-            this.shouldFetchDataElementsForProgram()
-        ) {
+    componentDidUpdate(_, prevState) {
+        const isDoneLoading =
+            prevState.loading === true && this.state.loading === false;
+        const hasRelevantPropChanged = this.hasRelevantPropChanged();
+        const requiresDataElementsForProgram = this.requiresDataElementsForProgram();
+        const shouldFetch =
+            hasRelevantPropChanged && requiresDataElementsForProgram;
+        const shouldSetOptions =
+            (isDoneLoading || hasRelevantPropChanged) &&
+            !requiresDataElementsForProgram;
+
+        if (shouldFetch) {
             this.fetchDataElements();
         }
 
+        if (shouldSetOptions) {
+            const options = this.getOptions();
+            const isCurrentValueInOptions = options.some(
+                ({ value }) => value === this.props.value
+            );
+
+            // Reset selection to default if selected option is no longer valid
+            if (!isCurrentValueInOptions) {
+                this.props.onChange({ target: { value: undefined } });
+            }
+
+            this.setState({ options });
```

Comment:
Is this used only for the component do update? If so I think a comment explaining that is warranted.